src/widgetastic_patternfly/__init__.py

```diff
@@ -1849,3 +1850,179 @@ def click_location(self, name, handle_alert=True):
+
+        # check readonly or editable
+        date.is_readonly
+        # fill current date
+        date.fill(datetime.now())
+        # read selected date for DatePicker
+        date.read()
+    """
+    textbox = TextInput(locator=Parameter('@locator'))
+
+    def __init__(self, parent, id=None, name=None, locator=None,
+                strptime_format='%m/%d/%Y', logger=None):
+        View.__init__(self, parent=parent, logger=logger)
+
+        self.strptime_format = strptime_format
+        base_locator = './/*[(self::input or self::textarea) and @{}={}]'
+
+        if id:
+            self.locator = base_locator.format('id', quote(id))
+        elif name:
+            self.locator = base_locator.format('name', quote(name))
+        elif locator:
+            self.locator = locator
+        else:
+            raise TypeError('You need to specify either, id, name or locator for DatePicker')
+
+    class HeaderView(View):
+        prev_button = Text(".//*[contains(@class, 'prev')]")
+        next_button = Text(".//*[contains(@class, 'next')]")
+        datepicker_switch = Text(".//*[contains(@class, 'datepicker-switch')]")
+        _elements = {}
+
+        def select(self, value):
+            for el, web_el in self._elements.items():
+                if el == value:
+                    web_el.click()
+                    return True
+
+        @property
+        def active(self):
+            for el, web_el in self._elements.items():
+                if bool(self.browser.classes(web_el) & {'active', 'focused'}):
+                    return el
+
+    @View.nested
+    class date_pick(HeaderView):    # noqa
+        DATES = ".//*[contains(@class, 'datepicker-days')]/table/tbody/tr/td"
+
+        @property
+        def _elements(self):
+            dates = {}
+            for el in self.browser.elements(self.DATES):
+                if not bool({'old', 'new', 'disabled'} & self.browser.classes(el)):
+                    dates.update({int(el.text): el})
+            return dates
+
+    @View.nested
+    class month_pick(HeaderView):   # noqa
+        MONTHS = ".//*[contains(@class, 'datepicker-months')]/table/tbody/tr/td/*"
+
+        @property
+        def _elements(self):
+            months = {}
+            for el in self.browser.elements(self.MONTHS):
+                if not bool({'disabled'} & self.browser.classes(el)):
+                    months.update({el.text: el})
+            return months
+
+    @View.nested
+    class year_pick(HeaderView):    # noqa
+        YEARS = ".//*[contains(@class, 'datepicker-years')]/table/tbody/tr/td/*"
+
+        @property
+        def _elements(self):
+            years = {}
+            for el in self.browser.elements(self.YEARS):
+                if not bool({'old', 'new', 'disabled'} & self.browser.classes(el)):
+                    years.update({int(el.text): el})
+            return years
+
+        def select(self, value):
+            for _yr in range(10):
```

Comment:
Why do you add such restriction ? 
As far as I can see widget allows you to choose any value.
Besides you can easily count number of transitions and get to right range. However, you test value after every click. I don't think this is optimal.